src/state/apis/nft/nftApi.ts

```diff
@@ -91,6 +116,30 @@ export const nftApi = createApi({
 
           return acc
         }, [])
+
+        const nftsById = data.reduce<NftState['nfts']['byId']>((acc, item) => {
+          if (!item.collection.id) return acc
+
+          const nftAssetId: AssetId = `${item.collection.id}/${item.id}`
+          acc[nftAssetId] = item
+
+          return acc
+        }, {})
+        const nftIds: AssetId[] = Object.keys(nftsById)
+
+        const collectionsById = data.reduce<NftState['collections']['byId']>((acc, item) => {
+          if (!item.collection.id) return acc
+
+          const collectionAssetId = item.collection.id
+          acc[collectionAssetId] = item.collection
+
+          return acc
+        }, {})
+        const collectionIds: AssetId[] = Object.keys(collectionsById)
+
+        dispatch(nft.actions.upsertCollections({ byId: collectionsById, ids: collectionIds }))
```

Comment:
I assume this will be moved to the collection query once NftItem is updated to use `collectionId` as the primary key?

Reply:
with just the nft user tokens, could also reduce to a single reduce

Reply:
We will still upsert collections here even when fully normalized on the PR above - the rationale is we do get some data from fetching items, and *possibly* more and/or better data from fetching collection/s separately, hence we do upsert collections alongside items

Reply:
This seems weird to me that the nft item req would provide additional collection details that the collection req wouldn't, but very well could be the case. What I didn't see here was any upsert on the `getNftCollection` query which made me think this would be changed up a bit in a follow up pr (ie. `getNftUserTokens` uses `upsertNfts`, while `getNftCollection` uses `upsertCollections`)